### PR TITLE
docs: add sample for getting started with BQML

### PR DESCRIPTION
**Pull request by DevStephanie**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: samples/snippets/bqml_getting_started_test.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    #
+    # Because the totals column is a STRUCT data type, we need to call
+    # Series.struct.field("transactions") to extract the transactions field.
+    # See the reference documentation below:
+    # https://cloud.google.com/python/docs/reference/bigframes/latest/bigframes.operations.structs.StructAccessor#bigframes_operations_structs_StructAccessor_field
+    transactions = df["totals"].struct.field("transactions")
+
+    # If the number of transactions is NULL, the value in the label
+    # column is set to 0. Otherwise, it is set to 1. These values
+    # represent the possible outcomes.
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:44]: Let's rearrange a bit. Remember: start with a high-level explanation before going into details.

```suggestion
    # The "label" values represent the outcome of the model's
    # prediction. In this case, the model predicts if there are any
    # ecommerce transactions within the Google Analytics session.
    # If the number of transactions is NULL, the value in the label
    # column is set to 0. Otherwise, it is set to 1.
```

File: samples/snippets/bqml_getting_started_test.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    label = transactions.notnull().map({True: 1, False: 0})
+
+    # Operating systems of users, extracting child field as a struct.
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:47]: Let's use the same explanation as https://cloud.google.com/bigquery/docs/create-machine-learning-model#query_details

Note: since we've already explained what `.struct.field(` is doing, we don't need to explain it again.

```suggestion
    # Extract the operating system of the visitor's device.
```

File: samples/snippets/bqml_getting_started_test.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    operatingSystem = df["device"].struct.field("operatingSystem")
+    operatingSystem = operatingSystem.fillna("")
+
+    # Indicates whether the users devices are mobile.
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:51]: Align with the text in the document. Also, use a verb to indicate what the code is actually doing. 

```suggestion
    # Extract whether the visitor's device is a mobile device.
```

**Review by tswast — CHANGES_REQUESTED** [review]

File: samples/snippets/bqml_getting_started_test.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        {"os": operatingSystem, "is_mobile": isMobile, "pageviews": pageviews}
+    )
+
+    # Logistic Regression model splits data into two classes, giving the
+    # probablity the data is in one of the classes.
+    model = LogisticRegression()
+    model.fit(features, label)
+
+    # When writing a DataFrame to a BigQuery table, include destinaton table
+    # and parameters, index defaults to "True".
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:71]: This comment has nothing to do with BigQuery ML models. Please fix.

Note: The important thing here is that we're taking our trained model and writing it to a permanent location.

File: samples/snippets/bqml_getting_started_test.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def test_bqml_getting_started():
+    # [START bigquery_getting_Started_bqml_tutorial]
```
[the remark this answers is not in the document]
Comment (DevStephanie) [reply samples/snippets/bqml_getting_started_test.py:17]: Agreed, corrected.

File: samples/snippets/bqml_getting_started_test.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    )
+
+    # Logistic Regression model splits data into two classes, giving the
+    # probablity the data is in one of the classes.
+    model = LogisticRegression()
+    model.fit(features, label)
+
+    # When writing a DataFrame to a BigQuery table, include destinaton table
+    # and parameters, index defaults to "True".
+    model.to_gbq("bqml_tutorial.sample_model", replace=True)
```
[the remark this answers is not in the document]
Comment (DevStephanie) [reply samples/snippets/bqml_getting_started_test.py:72]: Agreed, corrected.

File: samples/snippets/bqml_getting_started_test.py [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def test_bqml_getting_started():
+    # [START bigquery_getting_Started_bqml_tutorial]
+    from bigframes.ml.linear_model import LogisticRegression
+    import bigframes.pandas as bpd
+
+    # EXPLANATION - REFERENCE GBQ DOCS!
```
[the remark this answers is not in the document]
Comment (DevStephanie) [reply samples/snippets/bqml_getting_started_test.py:21]: Agreed, corrected.

File: samples/snippets/bqml_getting_started_test.py [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    df = bpd.read_gbq(
+        """
+        SELECT GENERATE_UUID() AS rowindex, *
```
[the remark this answers is not in the document]
Comment (DevStephanie) [reply samples/snippets/bqml_getting_started_test.py:24]: Agreed, corrected.

File: samples/snippets/bqml_getting_started_test.py (hunk 4 above)
Comment (DevStephanie) [reply samples/snippets/bqml_getting_started_test.py:71]: Agreed, corrected.

File: samples/snippets/bqml_getting_started_test.py (hunk 2 above)
Comment (DevStephanie) [reply samples/snippets/bqml_getting_started_test.py:47]: Agreed, corrected.

File: samples/snippets/bqml_getting_started_test.py (hunk 3 above)
Comment (DevStephanie) [reply samples/snippets/bqml_getting_started_test.py:51]: Agreed, corrected.

File: samples/snippets/bqml_getting_started_test.py (hunk 1 above)
Comment (DevStephanie) [reply samples/snippets/bqml_getting_started_test.py:58]: Agreed, corrected

File: samples/snippets/bqml_getting_started_test.py [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+=======
+    # [START bigquery_getting_Started_bqml_tutorial]
+>>>>>>> 9ec139e7e275e8082379022e3eb1ce06e9664c2e
+    from bigframes.ml.linear_model import LogisticRegression
+    import bigframes.pandas as bpd
+
+<<<<<<< HEAD
+    # Read_gbq loads a DataFrame from BiqQuery and gives an unordered, 
+    # unindexed data source. The default DataFrame will have an arbitary 
+    # index and ordering.
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:29]: Let's explain why we're using `read_gbq` as well as a bit of an explanation about the use of SQL here instead of just the table names. Also link to a feature request they can watch for when they won't have to use SQL anymore: https://github.com/googleapis/python-bigquery-dataframes/issues/169

```suggestion
    # Start by selecting the data you'll use for training. `read_gbq` accepts
    # either a SQL query or a table ID. Since this example selects from multiple
    # tables via a wildcard, use SQL to define this data. Watch issue
    # https://github.com/googleapis/python-bigquery-dataframes/issues/169
    # for updates to `read_gbq` to support wildcard tables.
```

File: samples/snippets/bqml_getting_started_test.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (tswast) [reply samples/snippets/bqml_getting_started_test.py:38]: You can use this suggestion as-is. Note that in SQL, `--` is how you make a comment.

File: samples/snippets/bqml_kmeans_clustering.py [hunk 10]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2023 Google LLC
```
Comment (tswast) [inline samples/snippets/bqml_kmeans_clustering.py:1]: We can remove this file for now. Let's do a separate PR for the K-Means tutorials.

**Review by tswast — CHANGES_REQUESTED** [review]

File: samples/snippets/bqml_kmeans_clustering.py (hunk 10 above)
Comment (DevStephanie) [reply samples/snippets/bqml_kmeans_clustering.py:1]: Corrected.

File: samples/snippets/bqml_getting_started_test.py [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def test_bqml_getting_started():
+<<<<<<< HEAD
+    # [start bigquery_getting_Started_bqml_tutorial]
```
[the remark this answers is not in the document]
Comment (DevStephanie) [reply samples/snippets/bqml_getting_started_test.py:18]: I don't know how I missed the "S" being upper case, but agreed and corrected.

File: samples/snippets/bqml_getting_started_test.py (hunk 9 above)
Comment (DevStephanie) [reply samples/snippets/bqml_getting_started_test.py:29]: I agree this is a better explanation of why we're using read_gbq in our code. Corrected

File: samples/snippets/bqml_getting_started_test.py [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def test_bqml_getting_started():
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (DevStephanie) [reply samples/snippets/bqml_getting_started_test.py]: Agreed, corrected and accepted changes.

File: samples/snippets/bqml_getting_started_test.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (tswast) [reply samples/snippets/bqml_getting_started_test.py:17]: Need to redo this change.

File: samples/snippets/bqml_getting_started_test.py [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+
+    # Logistic Regression model splits data into two classes, giving the
+    # probablity the data is in one of the classes.
+    model = LogisticRegression()
+    model.fit(features, label)
+
+    # When writing a DataFrame to a BigQuery table, include destinaton table
+    # and parameters, index defaults to "True".
+    model.to_gbq("bqml_tutorial.sample_model", replace=True)
+    # [END bigquery_getting_Started_bqml_tutorial]
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:85]: Let's keep region tag lower_case

```suggestion
    # [END bigquery_getting_started_bqml_tutorial]
```

File: samples/snippets/bqml_getting_started_test.py [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+    from bigframes.ml.linear_model import LogisticRegression
+    import bigframes.pandas as bpd
+
+    # Start by selecting the data you'll use for training. `read_gbq` accepts
+    # either a SQL query or a table ID. Since this example selects from multiple
+    # tables via a wildcard, use SQL to define this data. Watch issue
+    # https://github.com/googleapis/python-bigquery-dataframes/issues/169
+    # for updates to `read_gbq` to support wildcard tables.
+    #
+    # https://github.com/googleapis/python-bigquery-dataframes/issues/169
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:27]: Don't need to link to the issue twice. Remove this one.

```suggestion
```

File: samples/snippets/bqml_getting_started_test.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (tswast) [reply samples/snippets/bqml_getting_started_test.py:31]: Need to re-apply this change.

File: samples/snippets/bqml_getting_started_test.py (hunk 1 above)
Comment (tswast) [reply samples/snippets/bqml_getting_started_test.py:51]: Need to re-apply this suggestion. Also, let's update the description to reflect what's happening in the code.

```suggestion
    # The "label" values represent the outcome of the model's
    # prediction. In this case, the model predicts if there are any
    # ecommerce transactions within the Google Analytics session.
    # If the number of transactions is not NULL, the value in the label
    # column is set to 1. Otherwise, it is set to 0.
```

File: samples/snippets/bqml_getting_started_test.py [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+    # See the reference documentation below:
+    # https://cloud.google.com/python/docs/reference/bigframes/latest/bigframes.operations.structs.StructAccessor#bigframes_operations_structs_StructAccessor_field
+    transactions = df["totals"].struct.field("transactions")
+
+    # If the number of transactions is NULL, the value in the label
+    # column is set to 0. Otherwise, it is set to 1. These values
+    # represent the possible outcomes.
+    label = transactions.notnull().map({True: 1, False: 0})
+
+    # Choosing the operating system of the users devices.
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:54]: This verb tense is weird. Let's match the other fields and what's written at https://cloud.google.com/bigquery/docs/create-machine-learning-model#query_details

```suggestion
    # Extract the operating system of the visitor's device.
```

File: samples/snippets/bqml_getting_started_test.py [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+    operatingSystem = df["device"].struct.field("operatingSystem")
+    operatingSystem = operatingSystem.fillna("")
+
+    # Extract whether the visitor's device is a mobile device.
+    isMobile = df["device"].struct.field("isMobile")
+
+    # Extract where the visitors country of origin is.
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:61]: Let's match the SQL description.

```suggestion
    # Extract the country from which the sessions originated, based on the IP address.
```

File: samples/snippets/bqml_getting_started_test.py [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+    country = df["geoNetwork"].struct.field("country").fillna("")
+
+    # Total number of pageviews within the session.
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:64]:
```suggestion
    # Extract the total number of page views within the session.
```

File: samples/snippets/bqml_getting_started_test.py [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+    pageviews = df["totals"].struct.field("pageviews").fillna(0)
+
+    # Selecting values to represent data in columns in DataFrames.
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:67]: This comment needs to be updated with some information _why_ we're combining these columns.

```suggestion
    # Combine all the feature columns into a single DataFrame
    # to use as training data.
```

File: samples/snippets/bqml_getting_started_test.py (hunk 4 above)
Comment (tswast) [reply samples/snippets/bqml_getting_started_test.py:83]: FYI: The comment is still talking about _tables_ not _models_. I'll make a comment with a suggested edit.

File: samples/snippets/bqml_getting_started_test.py [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+        }
+    )
+
+    # Logistic Regression model splits data into two classes, giving the
+    # probablity the data is in one of the classes.
+    model = LogisticRegression()
+    model.fit(features, label)
+
+    # When writing a DataFrame to a BigQuery table, include destinaton table
+    # and parameters, index defaults to "True".
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:83]:
```suggestion
    # The model.fit() call above created a temporary model.
    # Use the to_gbq() method to write to a permanent location.
```

File: samples/snippets/bqml_kmeans_clustering.py (hunk 10 above)
Comment (tswast) [reply samples/snippets/bqml_kmeans_clustering.py:1]: This file still needs to be deleted.

**Review by tswast — CHANGES_REQUESTED** [review]

File: samples/snippets/bqml_getting_started_test.py [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+        WHERE
+        _TABLE_SUFFIX BETWEEN '20160801' AND '20170630'
+        """,
+        index_col="rowindex",
+    )
+
+    # Extract the total number of transactions within
+    # the Google Analytics session.
+    #
+    # Because the totals column is a STRUCT data type, we need to call
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:43]: Nit: We don't use "we" in our tech writing. go/second-person

```suggestion
    # Because the totals column is a STRUCT data type, call
```

File: samples/snippets/bqml_getting_started_test.py [hunk 21]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            "os": operatingSystem,
+            "is_mobile": isMobile,
+            "country": country,
+            "pageviews": pageviews,
+        }
+    )
+
+    # Logistic Regression model splits data into two classes, giving the
+    # probablity the data is in one of the classes.
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:81]: Logistic regression's output is indeed between 0 and 1 like a probability but it doesn't fully follow the rules of probability. https://stats.stackexchange.com/a/34936/6943

```suggestion
    # Logistic Regression model splits data into two classes, giving the
    # a confidence score that the data is in one of the classes.
```

**Review by tswast — APPROVED** [review]

File: samples/snippets/bqml_getting_started_test.py [hunk 22]
```diff
@@ -0,0 +1,88 @@
...
+    )
+
+    # Logistic Regression model splits data into two classes, giving the
+    # a confidence score that the data is in one of the classes.
+    model = LogisticRegression()
+    model.fit(features, label)
+
+    # The model.fit() call above created a temporary model.
+    # Use the to_gbq() method to write to a permanent location.
+    model.to_gbq("bqml_tutorial.sample_model", replace=True)
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:87]: FYI: We're getting

```
E           google.api_core.exceptions.BadRequest: 400 Concurrent update on same model: bigframes-dev:bqml_tutorial.sample_model is not supported. Share your usecase with the BigQuery DataFrames team at the [https://bit.ly/bigframes-feedback](https://www.google.com/url?q=https://bit.ly/bigframes-feedback&sa=D) survey.
```

failure in our test suite: https://fusion2.corp.google.com/invocations/8a7513c8-e7c9-4b5b-82fe-9a83c176fbc1/targets/bigframes%2Fpresubmit%2Fe2e/log

I think we'll need a test fixture for this to create a temporary place for the model and clean it up when the test finishes.

1. Create a file called `samples/snippets/conftest.py`.
2. In the `conftest.py` file you create, add a fixture called `random_model_id`, similar to this one: https://github.com/googleapis/python-bigquery/blob/f804d639fe95bef5d083afe1246d756321128b05/samples/snippets/conftest.py#L101-L111 except it'll call `delete_model(...)` instead of `delete_table(...)`.

    You'll also need to add "prefixer" https://github.com/googleapis/python-bigquery/blob/f804d639fe95bef5d083afe1246d756321128b05/samples/snippets/conftest.py#L21 and `bigquery_client` fixture https://github.com/googleapis/python-bigquery/blob/f804d639fe95bef5d083afe1246d756321128b05/samples/snippets/conftest.py#L33-L36
3. Update your code sample to use the new `random_model_id` fixture.

    Look how we do it in the remote functions test: https://github.com/googleapis/python-bigquery-dataframes/blob/81125f9505ad98e89939769a8e1fcf30518705f0/samples/snippets/remote_function.py#L17-L23 but instead you'll be setting `your_model_id = random_model_id` and calling

    ```
    model.to_gbq(
        your_model_id,  # "project.dataset.model_id" or "dataset.model_id"
        replace=True,
    )
    ```

**Review by tswast — CHANGES_REQUESTED** [review]

**Review by tswast — APPROVED** [review]

File: samples/snippets/bqml_getting_started_test.py [hunk 23]
```diff
@@ -0,0 +1,88 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def test_bqml_getting_started():
+    # [START bigquery_getting_started_bqml_tutorial]
```
Comment (tswast) [inline samples/snippets/bqml_getting_started_test.py:17]: Let's use `bigquery_dataframes_bqml_getting_started` for our region tags.

**Review by tswast — CHANGES_REQUESTED** [review]

File: samples/snippets/bqml_getting_started_test.py (hunk 23 above)
Comment (Sabdul96) [reply samples/snippets/bqml_getting_started_test.py:17]: Sounds great, will make edits now.

File: third_party/geopandas/LICENSE.txt [hunk 24]
```diff
@@ -0,0 +1,25 @@
+Copyright (c) 2013-2022, GeoPandas developers.
```
Comment (tswast) [inline third_party/geopandas/LICENSE.txt:1]: Revert this

**Review by tswast — APPROVED** [review]
